lib/completion.ts

```diff
@@ -34,6 +34,78 @@ export function completion(
     const current = args.length ? args[args.length - 1] : '';
     const argv = yargs.parse(args, true);
     const parentCommands = yargs.getContext().commands;
+
+    function defaultCompletion(): Arguments | void {
+      const handlers = command.getCommandHandlers();
```

Comment:
i would be tempted to pull each of these major pieces of logic into a helper function within `completion.ts`:

* `commandCompletions`?
* `optionCompletions`?
* `customCompletions`?